components/Breadcrumb/Breadcrumb.tsx

```diff
@@ -0,0 +1,152 @@
+import React, {
+  ReactNode,
+  ReactElement,
+  CSSProperties,
+  Component,
+  cloneElement,
+} from "react"
+import styled from "styled-components"
+import { BreadcrumbItemProps, StyledText } from "./BreadcrumbItem"
+
+export interface BreadcrumbProps {
+  /** separate to be using between crumbs */
+  separator?: string | ReactNode
+  /** Styles applied to the root */
+  rootStyle?: CSSProperties
+  /** Styles applied to each child */
+  childStyle?: CSSProperties
+  /** Breadcrumb Items in children */
+  children?: ReactNode[] | ReactNode
+  /** className for user to specify their class */
+  className?: string
+  /** Max width of the breadcrumb to be wrapped in */
+  maxWidth?: string
+  /** Level till which the crumbs are to be shown (from end). First is shown anyways */
+  truncateTo?: number
+  /** Styles for the active item */
+  activeStyles?: CSSProperties
+}
+
+const StyledParent: any = styled.div`
+  display: flex;
+  max-width: ${(props: any) => props.maxWidth};
+  flex-wrap: wrap;
+  width: fit-content;
+  align-items: center;
+`
+
+export default class Breadcrumb extends Component<BreadcrumbProps, any> {
```

Comment:
why do you need a class component?

Reply:
Because `.Item` property lives on the class.

Reply:
Defined `.Item` on FC initially but this is the way suggested by tsc.

Reply:
@Rohanhacker sensei 🙏